### PR TITLE
Add deprecation documentation

### PR DESCRIPTION
**Pull request by danielwilms**
Fixes: #216

### CODE REVIEW

**Review by SebastianSchildt — COMMENTED** [review]
I like it. As stated in discussion: Lets not overly complicate this in the first step. I consider my comments mere suggestions at the moment, I would even approve without them.
I think the mechanism with the `deprecation` metadata for nodes is sufficiently generic, that we can specify more semantics on a as-needed basis, once we hit some corner cases.

File: docs-gen/content/rule_set/basics.md [hunk 1]
```diff
@@ -71,3 +71,24 @@ be included as well, as shown below:
...
+* If the node is removed from the specification it shall be indicated by `removed` and optionally the reason for the removal as deprecation reason.
+* Nodes which are deprecated will be removed from the specification, either in the second minor update or, if earlier, the next major update.
+
+### Example
+```
+Navigation.CurrentLocation:
+  type: branch
+  aggregate: true
+  description: The current latitude and longitude of the vehicle.
+  deprecation: @2.1 moved to Vehicle.CurrentLocation
```
Comment (erikbosch) [inline docs-gen/content/rule_set/basics.md:93]: Is it always clear what the next release will be called, e.g. that next release actually will be called 2.1 in this example? Or do we need a temporary syntax in Git like "@X.Y" to use for the "next release" that is replaced with the actual release name as part of the release process?

**Comment by danielwilms** [comment]
Open points:
* Removed -> intent to remove
* clarify move to doesn't work with meta data changes
* recommend/indicate: server should keep deprecated nodes

File: docs-gen/content/rule_set/basics.md (hunk 1 above)
Comment (danielwilms) [reply docs-gen/content/rule_set/basics.md:94]: @danielwilms: add to the release process

**Review by gunnarx — CHANGES_REQUESTED** [review]
It looks like the format we suggest is not valid YAML :-(
Try pasting this into https://yamlvalidator.com/ 
```
- Navigation.CurrentLocation.Example:
  foo: bar
  deprecation: @2.1 moved to Vehicle.CurrentLocation.Longitude
``` 

The relevant specification is here - it states the @ sign is reserved:
https://yaml.org/spec/1.2/spec.html#c-reserved

The ability to use an off-the-shelf parser is critical IMHO.  It is the reason to use a standard format.  Therefore we need to come up with something different that is YAML compliant.

**Comment by gunnarx** [comment]
A proposal 
`   deprecation: V2.1 moved to Vehicle.CurrentLocation`

**Comment by SebastianSchildt** [comment]
Yes, but quotes are allowed in YAML

```
- Navigation.CurrentLocation.Example:
  foo: bar
  deprecation: "@2.1 moved to Vehicle.CurrentLocation.Longitude"
```

(If this is a"better" solution I do not know, but it is not a hack. Should validate)

**Pull request merged**